syncer/syncer.go

```diff
@@ -80,6 +81,8 @@ type Syncer struct {
 	fetcher  layerFetcher
 	syncOnce sync.Once
 	// access via atomic.[Load|Store]Uint32
+	started uint32
```

Comment:
is it used only to print a log line in case of stop?

Reply:
what happens is that `TestSpacemeshApp_NodeService` hung because it's trying to call syncer.Close() before syncer.Start() was executed. so syncer.Close() blocks on taskgroup.waitErr. 

do you think it make sense to add another way to terminate (explicitly) rather than waiting for one of its goroutines return error?

Reply:
oh thats another use case that i didn't anticipate. i think we can initialize waitErr only when first Go is called, and return Wait immediately if waitErr is nil